service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -106,13 +106,29 @@ public String getWsmServiceAccountToken() throws IOException {
    * @param userRequest - request object for this user
    * @return - email address of user
    */
-  public String getRequestUserEmail(AuthenticatedUserRequest userRequest)
+  public String getEmailFromRequestOrSam(AuthenticatedUserRequest userRequest)
       throws InterruptedException {
     Optional<String> emailMaybe = Optional.ofNullable(userRequest.getEmail());
     if (emailMaybe.isPresent()) {
```

Comment:
just `return emailMaybe.orElseGet(() -> getEmailFromSam(userRequest));`. You should rarely need to call `get()` directly. IJ should have an auto-fix hint for it.